File: frameworks/go/fiber/Implementation/parameters.go
package implementation

import (
	"github.com/gofiber/fiber/v3"
)

type ParametersOne struct {
	One int `uri:"one" json:"one"`
}

type ParametersTwo struct {
	One int `uri:"one" json:"one"`
	Two int `uri:"two" json:"two"`
}

// parametersRoutes bind router captures with c.Bind().URI, which converts each to the integer
// its field declares. Fiber tries routes in the order they are registered, so the static route
// comes before the capture that would also match its path.
func parametersRoutes(app *fiber.App, p *Payloads) {
	app.Get("/parameters/static/segment/literal", func(c fiber.Ctx) error { return c.JSON(&p.Small) })

	app.Get("/parameters/:one/segment/literal", func(c fiber.Ctx) error {
		var captured ParametersOne
		if err := c.Bind().URI(&captured); err != nil {
			return refuse(c, err)
		}
		return c.JSON(Echoed[ParametersOne]{&p.Small, captured})
	})

	app.Get("/parameters/:one/with-second/:two", func(c fiber.Ctx) error {
		var captured ParametersTwo
		if err := c.Bind().URI(&captured); err != nil {
			return refuse(c, err)
		}
		return c.JSON(Echoed[ParametersTwo]{&p.Small, captured})
	})
}

File: frameworks/python/sanic/Implementation/routes/parameters.py
from sanic import Blueprint
from sanic.response import json

from answers import echoed
from documented import Payload, answers
from models import Camel
from payloads import Payloads


class One(Camel):
    one: int


class Two(Camel):
    one: int
    two: int


class EchoedOne(Payload):
    echo: One


class EchoedTwo(Payload):
    echo: Two


def blueprint(p: Payloads) -> Blueprint:
    """parameters: path captures, which Sanic's router converts to the int each one declares. The
    router tries a static route before a capture, so the order they are added in does not matter."""
    routes = Blueprint("parameters")

    @routes.get("/parameters/static/segment/literal")
    @answers(Payload, "items.small")
    async def static(request):
        return json(p.small)

    @routes.get("/parameters/<one:int>/segment/literal")
    @answers(EchoedOne, "items.small, with the capture")
    async def one(request, one: int):
        return json(echoed(p.small, {"one": one}))

    @routes.get("/parameters/<one:int>/with-second/<two:int>")
    @answers(EchoedTwo, "items.small, with both captures")
    async def two(request, one: int, two: int):
        return json(echoed(p.small, {"one": one, "two": two}))

    return routes
